src/client/tile.cpp

```diff
@@ -1766,8 +1779,8 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			[applyfiltersformesh
 			Internal modifier
 		*/
-		else if (str_starts_with(part_of_name, "[applyfiltersformesh"))
-		{
+		// TODO: What should be done to normalmaps here?
```

Comment:
Looks like this modifier is there to make filtered textures appear sharper, so it's a good idea to rescale normal maps too.

Reply:
https://github.com/minetest/minetest/blob/040b878cd5f77f6b44622b7c00255b4981500dc8/minetest.conf.example#L480-L490

Normalmaps don't have sharpness, I think upscaling normal maps with nearest neighbour filter breaks them.
It doesn't seem to be applied on normalmaps, I've added an assertion.